### PR TITLE
chore(rustfmt): format let-chains and let-else

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Cherry-pick and test  https://github.com/rust-lang/rustfmt/pull/5203 and  https://github.com/rust-lang/rustfmt/pull/5690

Use this branch https://github.com/rust-lang/rustfmt/compare/master...xxchan:rustfmt:xxchan/cherry

install
```
git clone https://github.com/xxchan/rustfmt && cd rustfmt
git checkout xxchan/cherry
cargo build --release --bin rustfmt --bin cargo-fmt -Zunstable-options --out-dir "$(rustc --print=sysroot)/bin"
```

invoke
```
# add format_cargo_toml = true to rustfmt.toml and then
cargo +nightly-2023-01-24 fmt
```

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
haha `cargo sort` failed! :sob:

**Comment by xxchan** [comment]
format let-chains first, `Cargo.toml` next time!

File: src/frontend/src/optimizer/rule/join_project_transpose_rule.rs [hunk 1]
```diff
@@ -51,10 +51,22 @@ impl Rule for JoinProjectTransposeRule {
...
-            old_i2new_i = old_i2new_i.union(&join.i2l_col_mapping_ignore_join_type().composite(&project.o2i_col_mapping()));
+        let new_left = if let Some(project) = left.as_logical_project()
+            && left_input_index_on_condition
+                .iter()
+                .all(|index| project.exprs()[*index].as_input_ref().is_some())
+            && join_type != JoinType::RightAnti
+            && join_type != JoinType::RightSemi
+            && join_type != JoinType::RightOuter
+            && join_type != JoinType::FullOuter
+        {
```
Comment (xxchan) [inline src/frontend/src/optimizer/rule/join_project_transpose_rule.rs:62]: wow 😲 Is this our largest let chain?

File: src/frontend/src/optimizer/rule/pull_up_hop_rule.rs [hunk 2]
```diff
@@ -51,56 +51,85 @@ impl Rule for PullUpHopRule {
...
+                && left_input_index_on_condition.iter().all(|&index| {
+                    hop.output_window_start_col_idx()
+                        .map_or(true, |v| index != v)
+                        && hop.output_window_end_col_idx().map_or(true, |v| index != v)
+                })
+                && join_type != JoinType::RightAnti
+                && join_type != JoinType::RightSemi
+                && join_type != JoinType::RightOuter
+                && join_type != JoinType::FullOuter
+            {
```
Comment (xxchan) [inline src/frontend/src/optimizer/rule/pull_up_hop_rule.rs:65]: Oh, this seems larger 🥵

**Review by BugenZhao — APPROVED** [review]
LGTM 😄

**Comment by xxchan** [comment]
Should I wait for the holiday to merge it? 😄

**Review by TennyZhuang — APPROVED** [review]

**Pull request merged**